### PR TITLE
Build config optimisation

### PR DESCRIPTION
**Pull request by edloidas**
Fixed the tasks dependencies and execution order between `lib-admin-ui`  and `app-contentstudio`.
Refactored `clean` task definitions.

### CODE REVIEW

**Review by ashklianko — APPROVED** [review]
Couldn't fully check it because original issue is reproducing occasionally

File: build.gradle [hunk 1]
```diff
@@ -7,13 +7,16 @@ plugins {
     id 'org.aim42.htmlSanityCheck' version '1.1.6'
 }
 
+allprojects {
```
Comment (rymsha) [inline build.gradle]: why root needs it?

File: gradle/lib-admin.gradle [hunk 2]
```diff
@@ -25,7 +66,12 @@ ext {
     hasLibAdminUi = this.&hasLibAdminUi
 }
 
-task flush( type: Delete, dependsOn: clean ) {
-    description = 'Clean the project from built sources and dependencies'
-    delete '.xp'
+afterEvaluate {
+    if ( hasTask( 'clean' ) )
+    {
+        task flush( type: Delete, dependsOn: clean ) {
```
Comment (rymsha) [inline gradle/lib-admin.gradle]: flush is rudimental. clean should be sufficient. this is likely the reason we still observe the problem

**Comment by edloidas** [comment]
@rymsha Yup, my bad. I have tested everything again. And it seems the original issue was actually fixed by you already in #2964. 
Git failed to update my branch and I reproduced the problem on a stale code (when there were no `*-dev-resources` jars). 
As for the `flush` task, it wasn't the source of the original problem.
I've updated the code, keeping only small optimizations, that are still relevant. Build order problem is gone already.

**Review by rymsha — APPROVED** [review]

**Review by alansemenov — APPROVED** [review]

**Pull request merged**